src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/model/ccd/FinremCaseData.java

```diff
@@ -724,6 +724,18 @@ public boolean isAppAddressConfidential() {
         return YesOrNo.YES.equals(getContactDetailsWrapper().getApplicantAddressHiddenFromRespondent());
     }
 
+    @JsonIgnore
+    public String getApplicantSolicitorPostcode() {
+        return isConsentedApplication()
+            ? getContactDetailsWrapper().getSolicitorAddress().getPostCode()
+            : getContactDetailsWrapper().getApplicantSolicitorAddress().getPostCode();
```

Comment:
Looks like potentially NPE when calling `getApplicantSolicitorAddress`

Reply:
+

Reply:
[edd9d7e](https://github.com/hmcts/finrem-case-orchestration-service/pull/1753/commits/edd9d7e8004a157413f8dd9dad2fa37b4cf4fb1c)

Reply:
What Ashley is saying is that if getContactDetailsWrapper().getSolicitorAddress() returns null then you will get a NPE due to the getPostCode() call. The same applies for the getContactDetails().getApplicantSolicitorAddress()

Reply:
[c76ba92](https://github.com/hmcts/finrem-case-orchestration-service/pull/1753/commits/c76ba927215850187445182fcc43f4d9bcef4345)